### ISSUE TITLE
source mapping url comments stop safari 

### ISSUE THREAD

**Issue by zhiwww** [yahoo/react-intl#158]
//# sourceMappingURL=react.js.map

comments in all files under folder 'lib' stops safari. a 404 error interrupts everything.

condition:
1/ using webpack
2/ safari 9.0 (11601.1.43)

**Comment by ericf** [comment]
See: #162

**Comment by sirrodgepodge** [comment]
Hey Eric, 

Love what you've made here!

Still getting this error in Safari though :(

[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (main.js.map, line 0)

(running in localhost, Safari 9.0.3, Macbook Pro, OSX El Capitan 10.11.3)

**Comment by sirrodgepodge** [comment]
Also getting this:

ReferenceError: No locale data has been provided for this object yet.

Once more only in Safari, perhaps related to the "map" issue as I know for sure that the locale data has been added

**Comment by sirrodgepodge** [comment]
This may be the cause actually if the fix you mentioned above was to comment out the sourcemaps link (looks like Safari still evaluates commented out sourcemap links):
https://github.com/thlorenz/convert-source-map/issues/31

**Comment by sirrodgepodge** [comment]
Whew, k so the sourcemap 404 still happens but it's not happening in production so all good.

In case anyone stumbles across this the real issue here is that for the safari "intl" polyfill you actually need to load in individual languages.  Here is a simmed down version of my final code:

    import {addLocaleData} from 'react-intl';
    import enabledLanguages from './enabledLanguages';

    // use Intl polyfill if not supported (in Safari)
    if (!window.Intl) {
        window.Intl = require('intl');
        window.Intl.polyFill = true;
    }

    const localizations = {};
    enabledLanguages.forEach((val) => {
        localizations[val] = require(`../../localization/${val}`); // if JSONs get to big, stop requiring/bunding in all languages and switch to using API
    window.Intl.polyFill && require(`intl/locale-data/jsonp/${localizations[val].locale}`);
        localizations[val].messages = flattenMessages(localizations[val].messages);
        addLocaleData(require(`react-intl/lib/locale-data/${localizations[val].locale}`));
    });

    // use this if we want to allow nested messages
    function flattenMessages(nestedMessages = {}, prefix = '') {
        return Object.keys(nestedMessages).reduce((messages, key) => {
            let value       = nestedMessages[key];
            let prefixedKey = prefix ? `${prefix}.${key}` : key;

            if (typeof value === 'string') {
                messages[prefixedKey] = value;
            } else {
                objectAssign(messages, flattenMessages(value, prefixedKey));
            }

            return messages;
        }, {});
    }

    export default localizations;

**Comment by ericf** [comment]
The source map not loading is likely orthogonal and not causing the actual code int he app to fail. I'm not the owner of the Intl polyfill, but your of course free to open an PR to improve it.

**Comment by sirrodgepodge** [comment]
Hey Eric,

Thank you for your quick response!  Source maps is indeed orthogonal.

What I am proposing is a PR to react-intl that would:

#1: Make IntlProvider automatically include the Safari-required 'Intl' polyfill (at least by default, perhaps a flag could be passed as a prop to exclude if some odd user doesn't care about Safari working)

#2: Include adding the locale-specific polyfills (the real doozy, kept thinking I was missing something in react-intl itself) as part of the "addLocaleData" function

This might take me a while since I'm relatively new to React and you're doing some pretty nifty stuff so I want to get confirmation from you first.

Best,
Roger

**Comment by ericf** [comment]
> Make IntlProvider automatically include the Safari-required 'Intl' polyfill (at least by default, perhaps a flag could be passed as a prop to exclude if some odd user doesn't care about Safari working)

This isn't the approach that should be taken, React Intl assumes the `Intl` built-in. Use http://cdn.polyfill.io/v2/docs/ for automatic `Intl` polyfilling for Safari.

> Include adding the locale-specific polyfills (the real doozy, kept thinking I was missing something in react-intl itself) as part of the "addLocaleData" function

I don't understand this one.

**Comment by sirrodgepodge** [comment]
k, fair enough on point #1, though I think it should be made very obvious in the docs (as opposed to having to search through issues) that Safari requires the "intl" npm as a polyfill to work

explaining #2, the thing that confused me for a while is that once I had added the "intl"  npm polyfill I actually still needed to add each specific language that I wanted to use from this polyfill as well as those same specific languages from react-intl

Either I'm a total noob or this has tripped up other people too.

**Comment by Dindaleon** [comment]
I tried @sirrodgepodge solution but I am still getting the `failed to load error: unsupported URL` error.

**Comment by sirrodgepodge** [comment]
Hey Dindaleon, just saw this, please post code if you could, would like to help

My first guess is that one of your included languages is named inconsistently with how it is named in the Intl package, e.g. "rs" instead of "ru" for Russian or something like that.

I've also updated my snippets above to include a sample localization file as I realize it was incomplete without.

**Comment by Dindaleon** [comment]
@sirrodgepodge I got my repo here: github.com/Dindaleon/hapi-react-starter-kit if you want to take a look; however, I have not updated it with your suggestions. I did try a solution provided in issue https://github.com/yahoo/react-intl/issues/71 and seems to be working fine. There, they talk about this webpack plugin that is striping out the source map comments (https://www.npmjs.com/package/strip-sourcemap-loader) I Enabled it for my production build. 

I am still waiting for a react-intl v2 update, as Eric said it was going to be fixed for next update.
